src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs

```diff
@@ -12,36 +13,28 @@ namespace Microsoft.DotNet.NativeWrapper
     public static partial class Interop
     {
         public static readonly bool RunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        static readonly string HostFxrPath;
 
         static Interop()
         {
-            if (RunningOnWindows)
-            {
-                PreloadWindowsLibrary("hostfxr.dll");
-            }
+            HostFxrPath = (string)AppContext.GetData("HOSTFXR_PATH");
+            NativeLibrary.SetDllImportResolver(typeof(Interop).Assembly, HostFxrDllImportResolver);
         }
 
-        // MSBuild SDK resolvers are required to be AnyCPU, but we have a native dependency and .NETFramework does not
-        // have a built-in facility for dynamically loading user native dlls for the appropriate platform. We therefore 
-        // preload the version with the correct architecture (from a corresponding sub-folder relative to us) on static
-        // construction so that subsequent P/Invokes can find it.
```

Comment:
This comment makes it sound like this class will be used in .NET Framework. From the setup authoring for VS, `Microsoft.DotNet.NativeWrapper.dll` and copies of `hostfxr` (in x64/x86 subdirectories) seem to be dropped in PrivateAssemblies. Does this pre-loading need to stay?

I don't know enough about the SDK and how this class is used, so I'll leave this to @sfoslund to comment on.

Reply:
Yes - this runs on .NET Framework as well - in fact up until `dotnet sdk check` this code was only used on .NET Framework (as far as I can tell). So we will need to ifdef around the new stuff correctly.

Reply:
Ah - then the pre-loading of the copy of hostfxr needs to stay too.